### ISSUE TITLE
TextArea and TextInput seems not be controlled by "value" prop

### ISSUE THREAD

**Issue by ReactGirl** [arturbien/React95#13]
After submitting the form and try to reset the value, the input field is not updated to an empty string.

When the ***Publish it!*** button is clicked the following function is called:
```jsx harmony
const publishMessage = () => {
    addMessage(text).then(() => {
      dispatch(newMessage(text))
      setText(' ') // this does not clear the input
    });
  };
```
The state is updated to empty string, however it does not update the text area to empty string, which remains with the old values.

The function is triggered by the following code:

```jsx harmony
  <div>
      <h3>got something to say? </h3>
      <div
        style={{
          padding: "5px 0",
          background: "#ced0cf"
        }}
      >
        <TextArea
          value={text}
          onChange={e => updateText(e)}
          placeholder="Type in here.."
          onKeyPress={handleKeyPress}
        />
        <ButtonBar>
          <Button onClick={publishMessage}>Publish it!</Button>
        </ButtonBar>
      </div>
    </div>
```
Using normal html input field it works.

**Comment by anton-g** [comment]
I ran into the same thing. Would be good to be able to use in both controlled and uncontrolled mode. One idea would be that setting the `value` prop means you want to control it yourself, and leaving it empty would work like it does right now. Might have to add a `defaultValue` prop or similar to support setting an initial value though.

**Comment by arturbien** [comment]
@ReactGirl  @anton-g ok it's fixed in 1.0.3 🚀 use 'value' in controlled component, and 'defaultValue' in uncontrolled one 😊 (just like in React [docs](https://reactjs.org/docs/uncontrolled-components.html))
